slangpy/tests/slangpy_tests/test_torchintegration.py

```
@@ -727,5 +727,271 @@ def test_zero_size_dispatch(device_type: DeviceType):
+    """Passing a non-tensor to create_torch_tensor_marshall raises ValueError."""
+    layout = _get_layout(device_type)
+    with pytest.raises(ValueError, match="unsupported"):
+        ttm.create_torch_tensor_marshall(layout, "not a tensor")
+
+
+@pytest.mark.parametrize("device_type", DEVICE_TYPES)
+def test_factory_unsupported_torch_dtype_raises(device_type: DeviceType):
+    """Passing a tensor with unsupported dtype raises ValueError."""
+    layout = _get_layout(device_type)
+    t = torch.tensor([1.0 + 2.0j], dtype=torch.complex64, device="cuda")
+    with pytest.raises(ValueError, match="[Uu]nsupported"):
```

Ruff (0.15.9)

[warning] Pattern passed to `match=` contains metacharacters but is neither escaped nor raw

(RUF043)